### PR TITLE
feat: Implement `div_rem` for `UnsignedInteger`

### PR DESCRIPTION
**Pull request by fmoletta**
# TITLE

## Description

Implements the `div_rem` method for `UnsignedInteger`. This method is necessary to implement the [api](https://github.com/starknet-io/types-rs/blob/main/crates/stark-felt/src/lib.rs) in the felt standarization crate 

## Type of change

Please delete options that are not relevant.

- [x] New feature
- [ ] Bug fix
- [ ] Optimization

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run

### CODE REVIEW

**Comment by MauroToscano** [comment]
Some more tests should be added to check Rem part is working, and even the division. It can be an easy division that's easy to check by hand

File: math/src/unsigned_integer/element.rs [hunk 1]
```diff
@@ -2599,4 +2717,11 @@ mod tests_u256 {
         b >>= 222;
         assert_eq!(a, b);
     }
+
+    #[test]
+    fn div_rem_works_1() {
```
Comment (MauroToscano) [inline math/src/unsigned_integer/element.rs:2722]: multiplying_and_dividing_for_number_is_number_with_remainder_0

File: math/src/field/fields/fft_friendly/stark_252_prime_field.rs [hunk 2]
```diff
@@ -50,6 +50,34 @@ impl FieldElement<Stark252PrimeField> {
...
+
+impl PartialOrd for FieldElement<Stark252PrimeField> {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        self.representative().partial_cmp(&other.representative())
+    }
+}
+
+impl Ord for FieldElement<Stark252PrimeField> {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.representative().cmp(&other.representative())
```
Comment (diegokingston) [inline math/src/field/fields/fft_friendly/stark_252_prime_field.rs:79]: Are we sure that cmp is the best way (in terms of performance) to do this? If I do not remember incorrectly, @Oppen found that in some cases this was slow

Comment (MauroToscano) [reply math/src/field/fields/fft_friendly/stark_252_prime_field.rs]: The derive was slow, this should be fine

File: math/src/unsigned_integer/element.rs [hunk 3]
```diff
@@ -2599,4 +2717,38 @@ mod tests_u256 {
...
+
+    #[test]
+    fn multiplying_and_dividing_for_number_is_number_with_remainder_0() {
+        let a = U256::from_u128(12678920202929299999999999282828);
+        let b = U256::from_u128(9000000000000);
+        assert_eq!((a * b).div_rem(&b), (a, U256::from_u64(0)));
+    }
+
+    #[test]
+    fn div_rem_works_1() {
```
Comment (MauroToscano) [inline math/src/unsigned_integer/element.rs:2729]:
```suggestion
    fn 8_div_rem_3_is_2_2) {
```

File: math/src/unsigned_integer/element.rs [hunk 4]
```diff
@@ -2599,4 +2717,38 @@ mod tests_u256 {
...
+        let a: UnsignedInteger<4> = U256::from_u64(8);
+        let b = U256::from_u64(3);
+        assert_eq!(a.div_rem(&b), (U256::from_u64(2), U256::from_u64(2)));
+    }
+
+    #[test]
+    fn div_rem_works_2() {
```
Comment (MauroToscano) [inline math/src/unsigned_integer/element.rs:2736]:
```suggestion
    fn 500721_div_rem_5_is_100144_1() {
```

File: math/src/unsigned_integer/element.rs [hunk 5]
```diff
@@ -2599,4 +2717,38 @@ mod tests_u256 {
...
+        let a = U256::from_u64(500721);
+        let b = U256::from_u64(5);
+        assert_eq!(a.div_rem(&b), (U256::from_u64(100144), U256::from_u64(1)));
+    }
+
+    #[test]
+    fn div_rem_works_3() {
```
Comment (MauroToscano) [inline math/src/unsigned_integer/element.rs:2743]:
```suggestion
    fn div_rem_works_with_big_numbers() {
```

**Review by MauroToscano — APPROVED** [review]

**Review by gabrielbosio — APPROVED** [review]

**Pull request merged**